app/PrismApplicationLoader.scala

```diff
@@ -1,41 +1,103 @@
+import play.api._
+import play.api.{Configuration, Mode}
+import play.api.routing.Router
+import conf.{AWS, DynamoConfiguration, FileConfiguration, Identity, PrismConfiguration}
 import com.amazonaws.auth.DefaultAWSCredentialsProviderChain
-import com.amazonaws.regions.{Regions, Region}
-import conf.{AWS, FileConfiguration, DynamoConfiguration, Identity}
-import play.api.ApplicationLoader.Context
-import play.api.{Mode, Configuration}
-import play.api.inject.guice.{GuiceApplicationBuilder, GuiceApplicationLoader}
-import utils.Logging
-
-class PrismApplicationLoader extends GuiceApplicationLoader() with Logging {
-
-  override protected def builder(context: Context): GuiceApplicationBuilder = {
-    val identity = {
-      context.environment.mode match {
-        case Mode.Prod => AWS.instance.identity
-        case _ => None
-      }
-    }.getOrElse(Identity("deploy", "prism", "DEV"))
+import com.amazonaws.regions.Regions
+import play.api.mvc.EssentialFilter
+import utils.{Lifecycle, Logging, ScheduledAgent}
 
-    log.info(s"Getting config for $identity")
+import scala.collection.mutable
+import scala.concurrent.Future
 
-    val extraConfigs = List(
-      DynamoConfiguration(
-        new DefaultAWSCredentialsProviderChain(),
-        Regions.EU_WEST_1,
-        identity
-      ),
-      FileConfiguration(identity)
-    )
+class PrismApplicationLoader extends ApplicationLoader {
+  private var components: PrismComponents = _
```

Comment:
Is there any particular reason to use a `var` here?

Reply:
TLDR: no

It's there because that's how it was done in the [example that I copied](https://github.com/playframework/play-samples/blob/2.8.x/play-scala-compile-di-example/app/MyApplicationLoader.scala) but looks like the [Play docs](https://www.playframework.com/documentation/2.8.x/ScalaCompileTimeDependencyInjection) do without it so we can/should change

Reply:
✅